rustler/src/lib.rs

```diff
@@ -62,7 +63,13 @@ pub use crate::error::Error;
 pub mod r#return;
 pub use crate::r#return::Return;
 
+#[doc(hidden)]
+mod nif;
+pub use nif::Nif;
+
 pub type NifResult<T> = Result<T, Error>;
 
 #[cfg(feature = "derive")]
-pub use rustler_codegen::{NifMap, NifRecord, NifStruct, NifTuple, NifUnitEnum, NifUntaggedEnum};
+pub use rustler_codegen::{
+    init, nif, NifMap, NifRecord, NifStruct, NifTuple, NifUnitEnum, NifUntaggedEnum,
```

Comment:
Shouldn't `init!` and `nif!` be included unconditionally?

Reply:
Perhaps they should. But right now, we have `rustler_codegen` as an optional dependency behind the `derive` feature flag (which is default).

Maybe we don't need this as a feature at all? I think it was just put in place because previously `rustler_codegen` had to be pulled in manually by users and in the latest version, we pulled it in by default.

But now that `init` and `nif` are primary macros, I think we should probably remove it.

Reply:
Yep, can be done in a separate PR, though.